wagtail/wagtailsearch/backends/elasticsearch.py

```diff
@@ -398,6 +399,38 @@ def __repr__(self):
 
 
 class ElasticsearchSearchResults(BaseSearchResults):
+    def __init__(self, *args, **kwargs):
+        super(ElasticsearchSearchResults, self).__init__(*args, **kwargs)
+        self.highlight_params = {}
+
+    def highlight(self, **kwargs):
```

Comment:
Have you considered an API, where each call to ``.highlight()`` adds a single field?

For example:

```python
Page.objects.search("foo").highlight("body", param="value").highlight("title")
```

This would give us some runtime type checking from Python.